lib/Base/IO.cpp

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "glow/Base/IO.h"
+
+#include <cstdio>
+#include <sys/stat.h>
+#include <sys/types.h>
+
+namespace {
+
+static size_t filesize(std::string filename) {
+  struct stat st;
+  if (stat(filename.c_str(), &st) != 0) {
+    GLOW_ASSERT(false && "stat failed");
+  }
+  return (size_t)st.st_size;
+}
+
+} // namespace
+
+namespace glow {
+
+void writeToFile(const Tensor &T, std::string filename) {
+  FILE *fp = fopen(filename.c_str(), "wb");
+  GLOW_ASSERT(fp);
+  fwrite(T.getUnsafePtr(), T.getType().getElementSize(), T.size(), fp);
+  fclose(fp);
+}
+
+void readFromFile(Tensor &T, std::string filename) {
```

Comment:
I wonder if LLVM has any utils that we can use here.

Reply:
I started looking and it seems like `MemoryBuffer` is the way to read files into memory in LLVM?  But I got a bit worried about going down a rabbit hole here since I'm only so familiar with LLVMs APIs :-/.  Happy to have someone who knows them better land some improvements.  `fread` is kind of mid-90s,  I know :-p

Reply:
I think that the 3-line implementation that uses fread is probably better because it won't break when we upgrade to a new LLVM. 

Everything looks great. ship it.